src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -595,8 +595,6 @@ void FileSegment::completeBasedOnCurrentState(std::lock_guard<std::mutex> & cach
         case State::DOWNLOADED:
         {
             chassert(getDownloadedSizeUnlocked(segment_lock) == range().size());
-            assert(is_downloaded);
-            assert(!cache_writer);
```

Comment:
why remove them? I did not see them fail before? Is it because of the recent changes in temporary filers PR?

Reply:
No, it's not related to those changes, it can be reproduced by running unit tests with debug build (no such run in our CI). But it seems that there should be more proper solution, so let's hold this PR